service/staging/staging_controller.py

```diff
@@ -1,13 +1,77 @@
 import os
 
-from flask import Blueprint, jsonify
-
+from flask import Blueprint, jsonify, request, abort
+from werkzeug.utils import secure_filename
 
 staging_controller = Blueprint('staging', __name__)
 
 staging_dir = os.environ['STAGING_DIR']
 
+allowed_extensions = ['xml', 'pdf', 'tif', 'tiff', 'json']
+
 
 @staging_controller.route('', methods=['GET'], strict_slashes=False)
 def list_staging():
+    """
+    Lists files in the staging area.
+
+    :return list(str):
+    """
     return jsonify(os.listdir(staging_dir))
+
+
+@staging_controller.route('', methods=['POST'], strict_slashes=False)
+def upload_to_staging():
+    """
+    Uploads files to the staging area.
+
+    The upload endpoint is able to handle single files that were posted
+    under the 'file' key as well as multiple files at once that have been
+    posted under the 'files' key.
+
+    Returns HTTP status code 415 if one of the files' extension is not allowed.
+
+    Returns HTTP status code 400 if neither 'file' nor 'files' contained valid
+    file data.
+
+    :return dict:
```

Comment:
Immernoch unklar was hier returned wird. Aufgrund der Beschreibung darüber würde ich denken ein String oder int mit dem Statuscode und kein dict...

Reply:
Guter Punkt, ich hab mich auch schwer getan, das genauer zu definieren weil hier auf das Verhalten von Flask zurückgegriffen wird und streng genommen dokumentiert werden müsste, dass hier entweder ein String oder ein Tupel aus String und int zurückgegeben wird. Daraus baut Flask dann entsprechende Response-Objekte (siehe http://flask.pocoo.org/docs/1.0/quickstart/#about-responses).

Da es aber bei Benutzung der API in dem Fall nicht so sehr darauf ankommt, was die Funktion zurückliefert (die wird ja sowieso nur von Flask intern aufgerufen), sondern, was per HTTP als Response zurückkommt, hatte ich versucht, zumindest das JSON-Objekt etwas näher zu beschreiben. Rein technisch gesehen ist das natürlich nicht ganz korrekt.

Vorschlag: Wir lassen den Return-Type bei Controller-Funktionen weg, beschreiben aber den Response-Body im Text dahinter.

Reply:
OK Verstehe. Aber könnte man nicht trotzdem diese Erklärung von dir kurz zusammengefasst da hinschreiben? Also dass es halt nicht eindeutig ist.

Reply:
Und das bei jeder Funktion, die ein Flask-Endpoint ist?

Ein bisschen Wissen über die eingesetzten Frameworks sollte man meines Erachtens schon voraussetzen dürfen.